modules/data.land/R/match_species_id.R

```diff
@@ -0,0 +1,65 @@
+#' Match BETY species ID.
+#'
+#' Parses species codes in input data and matches them with the BETY species ID.
+#' 
+#' @param input_codes Character vector of species codes
+#' @param format_name Species code format name (character, length 1)
+#' @param betydb \code{dplyr} \code{src} object containing BETY connection 
```

Comment:
They are not the same thing, which is why this is not called `con`. A `dplyr` `src` object (as documented here) is a list containing a `con` as well as some information about what kind of connection it is. Building an `src` from a `con` object is complicated, but extracting a `con` from an `src` is straightforward (just `src$con`). I thought we had discussed moving towards using `dplyr` for database queries, which is why I designed it this way.